geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java

```diff
@@ -670,7 +670,13 @@ public void suspect(ID mbr, String reason) {
   @Override
   public boolean checkIfAvailable(ID mbr, String reason,
       boolean initiateRemoval) {
-    if (membersInFinalCheck.contains(mbr)) {
+    return checkIfAvailable(mbr, reason, initiateRemoval, true);
+  }
+
+  @Override
+  public boolean checkIfAvailable(ID mbr, String reason,
+      boolean initiateRemoval, boolean checkIfMemberInFinalCheck) {
```

Comment:
suggest renaming parameter `checkIfMemberInFinalCheck` to `assumeMembersInFinalCheckAreAvailable` since the latter makes more sense to callers who haven't read the internals of the method